aci/data_source_aci_interface_config.go

```diff
@@ -0,0 +1,95 @@
+package aci
+
+import (
+	"context"
+	"fmt"
+	"strconv"
+	"strings"
+
+	"github.com/ciscoecosystem/aci-go-client/v2/client"
+	"github.com/ciscoecosystem/aci-go-client/v2/models"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+)
+
+func dataSourceAciInterfaceConfiguration() *schema.Resource {
+	return &schema.Resource{
+		ReadContext:   dataSourceAciInterfaceConfigurationRead,
+		SchemaVersion: 1,
+		Schema: AppendBaseAttrSchema(AppendNameAliasAttrSchema(map[string]*schema.Schema{
+			"node": {
+				Type:         schema.TypeInt,
+				Required:     true,
+				ValidateFunc: validation.IntBetween(101, 4000),
+			},
+			"interface": {
+				Type:             schema.TypeString,
+				Required:         true,
+				ValidateDiagFunc: validateInterface,
+			},
+			"port_type": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ValidateFunc: validation.StringInSlice([]string{
+					"access",
+					"fabric",
+				}, false),
+				Default: "access",
+			},
+			"role": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+			},
+			"policy_group": {
+				Type:          schema.TypeString,
+				Optional:      true,
+				Computed:      true,
+				ConflictsWith: []string{"breakout"},
+			},
+			"breakout": {
+				Type:          schema.TypeString,
+				Optional:      true,
+				Computed:      true,
+				ConflictsWith: []string{"policy_group"},
+			},
+			"admin_state": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+			},
+			"pc_member": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+			},
+			"description": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+			},
+		})),
+	}
+}
+
+func dataSourceAciInterfaceConfigurationRead(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+	aciClient := m.(*client.Client)
+	node := strconv.Itoa(d.Get("node").(int))
+	portType := d.Get("port_type").(string)
+	parsedInterface := strings.Split(getInterfaceVal(d.Get("interface")), "/")
+	rn := fmt.Sprintf(models.RnInfraPortConfig, node, parsedInterface[0], parsedInterface[1], parsedInterface[2])
```

Comment:
How does this react when interface is 1/1 and not 1/1/1 ?

Reply:
getInterfaceVal - function always returns 3 elements, if we pass 1/1 we get 1/1/0.